build failure: python3Packages.pdf-plumber unit tests fail due to error in python3Packages.pdfminer-six

### Steps To Reproduce

Steps to reproduce the behavior:
1. Build `python3Packages.pdf-plumber`.
2. Observe the following errors:
```sh
error: builder for '/nix/store/w7pdx4lww37a02r5adqz7995snf2nblv-python3.12-pdfplumber-0.11.3.drv' failed with exit code 1;
       last 10 log lines:
       > >       mediabox_params: List[Any] = [
       >             resolve1(mediabox_param) for mediabox_param in self.attrs["MediaBox"]
       >         ]
       > E       TypeError: 'PDFObjRef' object is not iterable
       >
       > /nix/store/w9s0k1122y818alyzvgyj3xlrbq0n0s9-python3.12-pdfminer-six-20240706/lib/python3.12/site-packages/pdfminer/pdfpage.py:63: TypeError
       > =========================== short test summary info ============================
       > FAILED tests/test_basics.py::Test::test_annots - TypeError: 'PDFObjRef' object is not iterable
       > FAILED tests/test_basics.py::Test::test_annots_rotated - TypeError: 'PDFObjRef' object is not iterable
```

### Additional context

This is due to a bug in pdfminer-six. With its six-month release cycle, its downstream derivations will continue to fail for a long time.

See https://github.com/pdfminer/pdfminer.six/issues/1004 


There is a patch available at https://github.com/dhdaines/pdfminer.six/commit/ad101c152c71431a21bfa5a8dbe33b3ba385ceec

### Notify maintainers

For `pdf-plumber`:
@happysalada @fabaff 

For `pdfminer-six`:
@psyanticy @marsam @dotlambda 

cc: @natsukium relative to jupyter-lab

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.47, NixOS, 24.11 (Vicuna), 24.11.20240828.71e91c4`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - channels(root): `"nixos-23.05"`
 - channels(sec): `""`
 - nixpkgs: `/nix/store/887hpp8a2i99n9jjwcvz6qkhhhqsvzkg-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/pdfminer-six/default.nix
+++ b/pkgs/development/python-modules/pdfminer-six/default.nix
@@ -2,6 +2,7 @@
   lib,
   buildPythonPackage,
   fetchFromGitHub,
+  fetchpatch2,
   cryptography,
   charset-normalizer,
   pythonOlder,
@@ -26,6 +27,13 @@ buildPythonPackage rec {
   };
 
   patches = [
+    # https://github.com/pdfminer/pdfminer.six/pull/1027
+    (fetchpatch2 {
+      name = "fix-dereference-MediaBox.patch";
+      url = "https://github.com/pdfminer/pdfminer.six/pull/1027/commits/ad101c152c71431a21bfa5a8dbe33b3ba385ceec.patch?full_index=1";
+      excludes = [ "CHANGELOG.md" ];
+      hash = "sha256-fsSXvN92MVtNFpAst0ctvGrbxVvoe4Nyz4wMZqJ1aw8=";
+    })
     (substituteAll {
       src = ./disable-setuptools-git-versioning.patch;
       inherit version;
@@ -51,6 +59,12 @@ buildPythonPackage rec {
   ];
 
   nativeCheckInputs = [ pytestCheckHook ];
+
+  disabledTests = [
+    # The binary file samples/contrib/issue-1004-indirect-mediabox.pdf is
+    # stripped from fix-dereference-MediaBox.patch.
+    "test_contrib_issue_1004_mediabox"
+  ];
 
   passthru = {
     tests = {

